exercises/flatten-array/src/example.h

```diff
@@ -0,0 +1,19 @@
+#ifndef FLATTEN_ARRAY_H
+#define FLATTEN_ARRAY_H
+
+#define ARRAY_SIZE(arr) (sizeof(arr)/sizeof(arr[0]))
+#define NONE 0
+
+enum item_type { UNDEF, null, ELEMENT, ARRAY };
+struct item {
+   int value;
+   struct item *next;
+   struct item *array_elements;
+   enum item_type type;
```

Comment:
By convention, members of structs are often declared largest to smallest in terms of type bit-size. This allows the compiler to naturally lay out the struct in an in the most efficient way it can (without doing packing) due to memory alignment and word size. It is not important w.r.t. to functionality, but may be interesting for you. Following that idiom here would result in :

```c
struct item {
    struct item *next;
    struct item *array_elements;
    enum item_type type;
    int value;
};
```